### PR TITLE
Collaborative effort to establish a definition of done

### LINKED ISSUE #54: Write a DoD
Write a DoD for Pillarbox.

### PR DESCRIPTION
**Pull request by jboix**
# Description

Closes #54

This is a collaborative effort among team members to establish clear criteria for completing tasks.

### CODE REVIEW

**Review by defagos — APPROVED** [review]

File: DEFINITION_OF_DONE.md [hunk 1]
```diff
@@ -0,0 +1,20 @@
+# Definition of Done (DoD)
+
+This document is a collaborative effort among team members to establish clear criteria for completing tasks. The DoD serves as a guide to ensure the quality and completeness of work within the team.
+
+To meet our standards, every task should adhere to the following criteria:
+
+- Code has been reviewed by at least one other team member.
+- Functionality works as expected and acceptance criteria are met
+- Coding standards are followed.
+- No critical issues or bugs are present.
```
Comment (amtins) [inline DEFINITION_OF_DONE.md:10]: I'm not so sure about this one:
> No critical issues or bugs are present.

Even with unit tests, bugs may still be present. Unit tests can be used to test for the presence of bugs, but not their absence. 🥲

File: DEFINITION_OF_DONE.md [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+- Unit tests pass successfully and the code successfully builds.
+- Code meets accessibility standards.
+- Code is well-documented.
+- README is updated with relevant information.
+- API documentation is updated.
+- Code meets performance standards.
```
Comment (amtins) [inline DEFINITION_OF_DONE.md:16]: What standards?

File: DEFINITION_OF_DONE.md [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+- Code has been checked for common security issues.
+- Sensitive information is handled securely.
+- Deployment scripts are up-to-date.
```
Comment (amtins) [inline DEFINITION_OF_DONE.md:19]: What does this mean?

File: DEFINITION_OF_DONE.md [hunk 4]
```diff
@@ -0,0 +1,20 @@
...
+- Deployment process is documented.
```
Comment (amtins) [inline DEFINITION_OF_DONE.md:20]: What if everything is automated?

File: DEFINITION_OF_DONE.md (hunk 1 above)
Comment (jboix) [reply DEFINITION_OF_DONE.md:10]: I agree. In my opinion `No critical issues are present` would be better, in a sense that would simply be : The lib is not crashing immediately.,

**Comment by amtins** [comment]
I agree with most of the points. Only the ones where I left a comment deserve a little iteration because I'm not sure I understood them.

File: DEFINITION_OF_DONE.md (hunk 4 above)
Comment (jboix) [reply DEFINITION_OF_DONE.md:20]: Maybe we could merge this sentence with the one before into something like : `Deployment scripts are up-to-date and documented`. The intention here being that not only "the code" is documented but also the process and scripts that support the delivery of the lib.

File: DEFINITION_OF_DONE.md (hunk 2 above)
Comment (jboix) [reply DEFINITION_OF_DONE.md:16]: We have decided to change this into : `No critical performance impact is present`.

File: DEFINITION_OF_DONE.md [hunk 5]
```diff
@@ -0,0 +1,20 @@
...
+- Functionality works as expected and acceptance criteria are met
+- Coding standards are followed.
+- No critical issues or bugs are present.
+- Unit tests pass successfully and the code successfully builds.
+- Code meets accessibility standards.
+- Code is well-documented.
+- README is updated with relevant information.
+- API documentation is updated.
+- Code meets performance standards.
+- Code has been checked for common security issues.
```
Comment (jboix) [inline DEFINITION_OF_DONE.md:17]: To remove, will be added in the future if needed.

File: DEFINITION_OF_DONE.md (hunk 1 above)
Comment (jboix) [reply DEFINITION_OF_DONE.md]: Has been changed to : `No critical issues are presen`.

**Review by waliid — APPROVED** [review]

**Review by MGaetan89 — APPROVED** [review]

**Review by amtins — APPROVED** [review]
LGTM!

**Pull request merged**